podcasts/PodcastListViewController+Search.swift

```diff
@@ -46,7 +46,11 @@ extension PodcastListViewController: UIScrollViewDelegate, PCSearchBarDelegate {
     func showSortOrderOptions() {
         let options = OptionsPicker(title: L10n.sortBy.localizedUppercase)
 
-        let sortOption = Settings.homeFolderSortOrder()
+        var sortOption = Settings.homeFolderSortOrder()
+        if !FeatureFlag.podcastsSortChanges.enabled, sortOption == .recentlyPlayed {
+            Settings.setHomeFolderSortOrder(order: .dateAddedNewestToOldest)
+            sortOption = Settings.homeFolderSortOrder()
+        }
```

Comment:
Would it make sense to keep `sortOption` immutable and set `homeFolderSortOrder` in the `else` block, just so that there's no confusion about it mutating further down?

```suggestion
        let sortOption: LibrarySort
        if !FeatureFlag.podcastsSortChanges.enabled && Settings.homeFolderSortOrder() == .recentlyPlayed {
            Settings.setHomeFolderSortOrder(order: .dateAddedNewestToOldest)
            sortOption = .dateAddedNewestToOldest
        } else {
            sortOption = Settings.homeFolderSortOrder()
        }
```